// @crossmint/client-sdk-react-ui/dist/chunk-XBXL7GMY.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { newObj[key] = obj[key]; } } } newObj.default = obj; return newObj; } } function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _react = require('react'); var c = _interopRequireWildcard(_react);var _inputotp = require('input-otp');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _react2 = require('@emotion/react');var _jsxruntime = require('react/jsx-runtime');var C=_react.createContext.call(void 0, void 0),O=_styled2.default.div`
    display: flex;
    align-items: center;
    gap: 8px;
    
    &:has(:disabled) {
        opacity: 0.5;
    }
`,T=_styled2.default.call(void 0, _inputotp.OTPInput)`
    &:disabled {
        cursor: not-allowed;
    }
`,v= exports.a =c.forwardRef((n,o)=>{var a=n,{appearance:e}=a,t=_chunk2WSEHVNPcjs.c.call(void 0, a,["appearance"]);return _jsxruntime.jsx.call(void 0, C.Provider,{value:e,children:_jsxruntime.jsx.call(void 0, O,{children:_jsxruntime.jsx.call(void 0, T,_chunk2WSEHVNPcjs.a.call(void 0, {autoFocus:!0,ref:o},t))})})});v.displayName="InputOTP";var k=_styled2.default.div`
    display: flex;
    gap: 8px;
    align-items: center;
`,$= exports.b =c.forwardRef((o,t)=>{var e=_chunk2WSEHVNPcjs.c.call(void 0, o,[]);return _jsxruntime.jsx.call(void 0, k,_chunk2WSEHVNPcjs.a.call(void 0, {ref:t},e))});$.displayName="InputOTPGroup";var w=_react2.keyframes`
    0%, 50% { opacity: 1; }
    51%, 100% { opacity: 0; }
`,E=_styled2.default.div`
    position: relative;
    display: flex;
    height: 56px;
    width: 48px;
    align-items: center;
    justify-content: center;
    border: 1px solid;
    font-size: 18px;
    transition: all 200ms ease;
    border-radius: ${e=>{var t;return((t=e.appearance)==null?void 0:t.borderRadius)||"6px"}};
    border-color: ${e=>{var t,o,n,a;return e.hasError?((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.danger)||_chunkEAWYV5ZOcjs.a["cm-danger"]:((a=(n=e.appearance)==null?void 0:n.colors)==null?void 0:a.border)||_chunkEAWYV5ZOcjs.a["cm-border"]}};
    background-color: ${e=>{var t,o,n,a;return e.hasChar?((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.buttonBackground)||_chunkEAWYV5ZOcjs.a["cm-muted-primary"]:((a=(n=e.appearance)==null?void 0:n.colors)==null?void 0:a.inputBackground)||_chunkEAWYV5ZOcjs.a["cm-background-primary"]}};
    color: ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    box-shadow: ${e=>{var t,o;return e.isActive?`0 0 0 2px ${((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}`:"none"}};
    z-index: ${e=>e.isActive?10:1};
`,S=_styled2.default.div`
    pointer-events: none;
    position: absolute;
    inset: 0;
    display: flex;
    align-items: center;
    justify-content: center;
`,A=_styled2.default.div`
    height: 18px;
    width: 1px;
    background-color: ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    animation: ${w} 1000ms infinite;
`,U= exports.c =c.forwardRef((a,n)=>{var l=a,{index:e,hasError:t}=l,o=_chunk2WSEHVNPcjs.c.call(void 0, l,["index","hasError"]);let P=c.useContext(_inputotp.OTPInputContext),{char:m,hasFakeCaret:x,isActive:y}=P.slots[e],u=_react.useContext.call(void 0, C);return _jsxruntime.jsxs.call(void 0, E,_chunk2WSEHVNPcjs.b.call(void 0, _chunk2WSEHVNPcjs.a.call(void 0, {ref:n,appearance:u,hasError:t,isActive:y,hasChar:!!m},o),{children:[m,x&&_jsxruntime.jsx.call(void 0, S,{children:_jsxruntime.jsx.call(void 0, A,{appearance:u})})]}))});U.displayName="InputOTPSlot";exports.a = v; exports.b = $; exports.c = U;
