// @crossmint/client-sdk-react-ui/dist/chunk-N2FP2XTI.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkS4BKL6MRcjs = require('./chunk-S4BKL6MR.cjs');var _chunkAHMPKIEYcjs = require('./chunk-AHMPKIEY.cjs');var _chunkDJZOP4MTcjs = require('./chunk-DJZOP4MT.cjs');var _chunk3X24JZVGcjs = require('./chunk-3X24JZVG.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunkNRN4WNQLcjs = require('./chunk-NRN4WNQL.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _react = require('react');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _color = require('color'); var _color2 = _interopRequireDefault(_color);var _jsxruntime = require('react/jsx-runtime');var z=_styled2.default.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    width: 100%;
`,A=_styled2.default.p`
    font-size: 16px;
    font-weight: 400;
    margin: 16px 0 4px 0;
    text-align: center;
    color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
`,D=_styled2.default.p`
    text-align: center;
    padding: 0 16px;
    margin: 0;
    color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.textSecondary)||_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    font-size: 14px;
    line-height: 1.4;
    overflow-wrap: anywhere;
`,L=_styled2.default.div`
    width: 100%;
    padding: 28px 0px 16px
`,N=_styled2.default.form`
    position: relative;
`,F=_styled2.default.input`
    ${_chunk3X24JZVGcjs.a}
    
    /* Base input styles */
    display: flex;
    flex-grow: 1;
    text-align: left;
    padding-left: 16px;
    padding-right: 80px;
    height: 58px;
    width: 100%;
    border: 1px solid;
    border-color: ${t=>{var o,e,s,l;return t.hasError?((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.danger)||_chunkEAWYV5ZOcjs.a["cm-danger"]:((l=(s=t.appearance)==null?void 0:s.colors)==null?void 0:l.border)||_chunkEAWYV5ZOcjs.a["cm-border"]}};
    border-radius: ${t=>{var o;return((o=t.appearance)==null?void 0:o.borderRadius)||"12px"}};
    background-color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.inputBackground)||_chunkEAWYV5ZOcjs.a["cm-background-primary"]}};
    color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    font-size: 16px;
    transition: border-color 200ms ease-in-out, box-shadow 200ms ease-in-out;
    
    &::placeholder {
        font-weight: 400;
        color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.textSecondary)||_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    }
    
    &:focus {
        outline: none;
        box-shadow: 0 0 0 2px ${t=>{var o,e;return new (0, _color2.default)(((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]).alpha(.18).toString()}};
    }
    
    &:read-only {
        cursor: not-allowed;
        opacity: 0.7;
    }
`,H=_styled2.default.div`
    position: absolute;
    right: 0;
    top: 0;
    bottom: 0;
    display: flex;
    align-items: center;
    padding-right: 16px;
    aspect-ratio: 1/1;
`,M=_styled2.default.button`
    ${_chunk3X24JZVGcjs.a}
    
    font-weight: 500;
    white-space: nowrap;
    background-color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.inputBackground)||_chunkEAWYV5ZOcjs.a["cm-background-primary"]}};
    color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}};
    cursor: ${t=>t.disabled?"not-allowed":"pointer"};
    opacity: ${t=>t.disabled?.6:1};
    border: none;
    padding: 8px 12px;
    border-radius: 6px;
    transition: opacity 200ms ease;
    
    &:hover:not(:disabled) {
        opacity: 0.8;
    }
    
    &:focus-visible {
        outline: 2px solid ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}};
        outline-offset: 2px;
    }

    &:not(:disabled):focus-visible {
        box-shadow: 0 0 0 2px ${t=>{var o,e;return new (0, _color2.default)(((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]).alpha(.18).toString()}};
    }
`,V=_styled2.default.div`
    font-size: 14px;
    color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.danger)||_chunkEAWYV5ZOcjs.a["cm-danger"]}};
    margin-bottom: 16px;
`,j=_styled2.default.div`
    font-size: 12px;
    line-height: 1.3;
    text-align: center;
    color: ${t=>{var o,e;return((e=(o=t.appearance)==null?void 0:o.colors)==null?void 0:e.textSecondary)||_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    margin-bottom: 16px;
`;function _({contactType:t,icon:o,title:e,description:s,onSubmitOTP:l,onResendCode:b,appearance:r,otpLength:h=9,helpText:O}){var I,$,S;let[m,R]=_react.useState.call(void 0, ""),[d,y]=_react.useState.call(void 0, !1),[C,g]=_react.useState.call(void 0, null),p=C!=null,w=m.length===h,U=()=>_chunk2WSEHVNPcjs.d.call(void 0, this,null,function*(){y(!0);try{yield l(m),g(null)}catch(c){console.error(`Error confirming ${t} OTP:`,c),g("Invalid code. Please try again.")}finally{y(!1)}});return _jsxruntime.jsxs.call(void 0, z,{children:[o,_jsxruntime.jsx.call(void 0, A,{appearance:r,children:e}),_jsxruntime.jsx.call(void 0, D,{appearance:r,children:s}),_jsxruntime.jsx.call(void 0, L,{children:_jsxruntime.jsxs.call(void 0, N,{role:"form",onSubmit:c=>{w&&(c.preventDefault(),U())},noValidate:!0,children:[_jsxruntime.jsx.call(void 0, _chunkS4BKL6MRcjs.a,{children:"OTP Code"}),_jsxruntime.jsx.call(void 0, F,{id:"otpInput",maxLength:h,appearance:r,hasError:p,type:"text",placeholder:"Enter code",value:m,onChange:c=>{R(c.target.value),g(null)},readOnly:d,"aria-describedby":"otpError"}),_jsxruntime.jsxs.call(void 0, H,{children:[p&&_jsxruntime.jsx.call(void 0, _chunkNRN4WNQLcjs.a,{customColor:(I=r==null?void 0:r.colors)==null?void 0:I.danger}),d&&_jsxruntime.jsx.call(void 0, _chunkAHMPKIEYcjs.a,{style:{color:($=r==null?void 0:r.colors)==null?void 0:$.textSecondary,fill:(S=r==null?void 0:r.colors)==null?void 0:S.textPrimary}}),!p&&!d?_jsxruntime.jsx.call(void 0, M,{type:"submit",appearance:r,disabled:!w||d,children:"Submit"}):null]})]})}),p&&_jsxruntime.jsx.call(void 0, V,{appearance:r,children:C}),_jsxruntime.jsx.call(void 0, j,{appearance:r,children:O}),b&&_jsxruntime.jsx.call(void 0, _chunkDJZOP4MTcjs.a,{initialSeconds:60,appearance:r,countdownText:c=>`Re-send code in ${c}s`,countdownCompleteText:"Re-send code",handleOnClick:b})]})}exports.a = _;
