// @crossmint/client-sdk-react-ui/dist/chunk-S5QCH2ED.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkHNUI5756cjs = require('./chunk-HNUI5756.cjs');var _chunk3X24JZVGcjs = require('./chunk-3X24JZVG.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunkEO53B7LLcjs = require('./chunk-EO53B7LL.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var f=_styled2.default.button`
    ${_chunk3X24JZVGcjs.a}
    
    position: absolute;
    top: 0px;
    left: 0px;
    width: 32px;
    height: 32px;
    border-radius: 6px;
    border: none;
    z-index: 1000;
    background-color: transparent;
    color: ${e=>{var t,o,r;return(r=(o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.textPrimary)!=null?r:_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 0.15s ease;
    
    &:hover {
        background-color: ${e=>{var t,o;return((o=(t=e==null?void 0:e.appearance)==null?void 0:t.colors)==null?void 0:o.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
        color: ${e=>{var t,o,r;return(r=(o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.textPrimary)!=null?r:_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    }
    
    &:focus {
        outline: none;
        background-color: ${e=>{var t,o;return((o=(t=e==null?void 0:e.appearance)==null?void 0:t.colors)==null?void 0:o.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
        box-shadow: 0 0 0 2px ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}};
    }
    
    svg {
        width: 24px;
        height: 24px;
    }
`,v= exports.a =y=>{var n=y,{iconColor:e,ringColor:t,onClick:o}=n,r=_chunk2WSEHVNPcjs.c.call(void 0, n,["iconColor","ringColor","onClick"]);let{setError:u,appearance:g}=_chunkHNUI5756cjs.a.call(void 0, );return _jsxruntime.jsx.call(void 0, f,_chunk2WSEHVNPcjs.b.call(void 0, _chunk2WSEHVNPcjs.a.call(void 0, {appearance:g,onClick:x=>{u(null),o==null||o(x)}},r),{children:_jsxruntime.jsx.call(void 0, _chunkEO53B7LLcjs.a,{style:{height:"24px",width:"24px",color:e}})}))};exports.a = v;
