// @crossmint/client-sdk-react-ui/dist/chunk-QVM6WPJU.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkSCHRLC2Icjs = require('./chunk-SCHRLC2I.cjs');var _chunkO3H2EYFAcjs = require('./chunk-O3H2EYFA.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunkQYQERFYLcjs = require('./chunk-QYQERFYL.cjs');var _chunkVI6TW7JTcjs = require('./chunk-VI6TW7JT.cjs');var _chunkTWX7ZBKZcjs = require('./chunk-TWX7ZBKZ.cjs');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var v=_styled2.default.div`
    position: relative;
    display: flex;
    flex-direction: column;
    gap: 10px;
    -webkit-font-smoothing: antialiased;
`,P=_styled2.default.div`
    display: flex;
    justify-content: center;
    position: relative;
    left: 6px;
`,b=_styled2.default.div`
    display: flex;
    justify-content: center;
`,w=_styled2.default.p`
    font-size: 18px;
    line-height: 28px;
    font-weight: bold;
    color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    margin: 0;
`,B=_styled2.default.div`
    margin-bottom: 24px;
`,T=_styled2.default.div`
    font-weight: normal;
    color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.textSecondary)||_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    line-height: 1.5;
`,I=_styled2.default.div`
    display: flex;
    flex-direction: column;
    gap: 16px;
    justify-content: center;
`,S=_styled2.default.div`
    display: flex;
    justify-content: center;
    padding-top: 16px;
`,x=_styled2.default.div`
    display: flex;
    flex-direction: column;
    gap: 8px;
`,d=_styled2.default.div`
    display: flex;
    gap: 8px;
    align-items: flex-start;
`,u=_styled2.default.div`
    flex-shrink: 0;
`,U=_styled2.default.div`
    margin-bottom: 12px;
`,A=_styled2.default.button`
    position: relative;
    display: flex;
    font-size: 16px;
    padding: 16px;
    background-color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.buttonBackground)||_chunkEAWYV5ZOcjs.a["cm-muted-primary"]}};
    color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    align-items: center;
    width: 100%;
    border-radius: ${o=>{var e;return((e=o.appearance)==null?void 0:e.borderRadius)||"12px"}};
    justify-content: center;
    transition: all 200ms ease;
    border: none;
    cursor: pointer;
    
    &:hover {
        background-color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
    }
    
    &:focus {
        background-color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
        outline: none;
    }
`,F=_styled2.default.span`
    font-weight: 500;
    margin: 0 32px;
    color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
`,$=_styled2.default.span`
    position: absolute;
    width: 1px;
    height: 1px;
    padding: 0;
    margin: -1px;
    overflow: hidden;
    clip: rect(0, 0, 0, 0);
    white-space: nowrap;
    border: 0;
`,L=_styled2.default.a`
    padding: 14px;
    text-align: center;
    text-decoration: none;
    border-radius: 8px;
    font-weight: bold;
    background-color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.inputBackground)||_chunkEAWYV5ZOcjs.a["cm-muted-primary"]}};
    color: ${o=>{var e,n;return((n=(e=o.appearance)==null?void 0:e.colors)==null?void 0:n.textSecondary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    transition: opacity 200ms ease;
    
    &:hover {
        opacity: 0.8;
    }
`;function p({title:o,content:e,primaryButton:n,secondaryAction:k,open:h,appearance:c}){var m;return _jsxruntime.jsx.call(void 0, _chunkO3H2EYFAcjs.a,{open:h,appearance:c,showCloseButton:!1,children:_jsxruntime.jsxs.call(void 0, v,{children:[_jsxruntime.jsx.call(void 0, P,{children:_jsxruntime.jsx.call(void 0, _chunkTWX7ZBKZcjs.a,{appearance:c})}),_jsxruntime.jsx.call(void 0, b,{children:_jsxruntime.jsx.call(void 0, w,{appearance:c,children:o})}),_jsxruntime.jsx.call(void 0, B,{children:_jsxruntime.jsx.call(void 0, T,{appearance:c,children:e})}),_jsxruntime.jsxs.call(void 0, I,{children:[n,k]}),_jsxruntime.jsx.call(void 0, S,{children:_jsxruntime.jsx.call(void 0, _chunkSCHRLC2Icjs.a,{color:(m=c==null?void 0:c.colors)==null?void 0:m.textSecondary})})]})})}var s=({appearance:o,onClick:e,children:n})=>_jsxruntime.jsxs.call(void 0, A,{appearance:o,onClick:e,children:[_jsxruntime.jsx.call(void 0, F,{appearance:o,children:n}),_jsxruntime.jsx.call(void 0, $,{children:n})]});function _({state:o,appearance:e}){if(!o.open||o.type==null)return null;switch(o.type){case"create-wallet":return _jsxruntime.jsx.call(void 0, p,{open:o.open,title:"Create Your Wallet",appearance:e,content:_jsxruntime.jsxs.call(void 0, _jsxruntime.Fragment,{children:[_jsxruntime.jsx.call(void 0, U,{children:"You're about to create a wallet."}),_jsxruntime.jsxs.call(void 0, x,{children:[_jsxruntime.jsxs.call(void 0, d,{children:[_jsxruntime.jsx.call(void 0, u,{children:_jsxruntime.jsx.call(void 0, _chunkVI6TW7JTcjs.a,{})}),"Your wallet will be secured with a passkey"]}),_jsxruntime.jsxs.call(void 0, d,{children:[_jsxruntime.jsx.call(void 0, u,{children:_jsxruntime.jsx.call(void 0, _chunkQYQERFYLcjs.a,{})}),"Your device will ask you for your fingerprint, face, or screen lock to set it up"]})]})]}),primaryButton:_jsxruntime.jsx.call(void 0, s,{appearance:e,onClick:o.primaryActionOnClick,children:"Create Wallet"})});case"create-wallet-error":return _jsxruntime.jsx.call(void 0, p,{open:o.open,title:"Wallet Creation Failed",appearance:e,content:_jsxruntime.jsx.call(void 0, "div",{children:"We couldn't create your wallet. This could be due to rejecting the request, a timeout, or not having access to your passkey on this device."}),primaryButton:_jsxruntime.jsx.call(void 0, s,{appearance:e,onClick:o.primaryActionOnClick,children:"Try again"})});case"transaction":return _jsxruntime.jsx.call(void 0, p,{open:o.open,title:"Use Your Wallet",appearance:e,content:_jsxruntime.jsx.call(void 0, x,{children:_jsxruntime.jsxs.call(void 0, d,{children:[_jsxruntime.jsx.call(void 0, u,{children:_jsxruntime.jsx.call(void 0, _chunkQYQERFYLcjs.a,{})}),_jsxruntime.jsx.call(void 0, "span",{children:"Your device will ask you for your fingerprint, face, or screen lock to authorize this action."})]})}),primaryButton:_jsxruntime.jsx.call(void 0, s,{appearance:e,onClick:o.primaryActionOnClick,children:"Use Wallet"})});case"transaction-error":return _jsxruntime.jsx.call(void 0, p,{open:o.open,title:"Wallet Access Failed",appearance:e,content:_jsxruntime.jsx.call(void 0, "div",{children:"We couldn't access your wallet. This could be due to rejecting the request, a timeout, or not having access to your passkey on this device."}),primaryButton:_jsxruntime.jsx.call(void 0, s,{appearance:e,onClick:o.primaryActionOnClick,children:"Try again"}),secondaryAction:_jsxruntime.jsx.call(void 0, L,{href:"https://docs.crossmint.com/wallets/signers-and-custody#passkey",rel:"noopener noreferrer",target:"_blank",appearance:e,children:"Troubleshoot"})});case"not-supported":return _jsxruntime.jsx.call(void 0, p,{open:o.open,title:"Passkeys Not Supported on This Device",appearance:e,content:_jsxruntime.jsx.call(void 0, "div",{children:"To access your wallet with a passkey, switch to a device or browser that supports passkeys, such as Chrome or Safari on a smartphone, tablet, or modern computer"}),primaryButton:_jsxruntime.jsx.call(void 0, s,{appearance:e,onClick:o.primaryActionOnClick,children:"Understood"})});default:return null}}exports.a = _;
