// @crossmint/client-sdk-react-ui/dist/chunk-ZNFOZDZO.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkHNUI5756cjs = require('./chunk-HNUI5756.cjs');var _chunkYCYSZLQ2cjs = require('./chunk-YCYSZLQ2.cjs');var _chunkS4BKL6MRcjs = require('./chunk-S4BKL6MR.cjs');var _chunk3X24JZVGcjs = require('./chunk-3X24JZVG.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunkCDQFTAM2cjs = require('./chunk-CDQFTAM2.cjs');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var f=_styled2.default.button`
    ${_chunk3X24JZVGcjs.a}
    
    /* Base button styles */
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 16px;
    font-size: 16px;
    min-height: 56px;
    border: none;
    border-radius: ${e=>{var t;return((t=e.appearance)==null?void 0:t.borderRadius)||"12px"}};
    background-color: ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.buttonBackground)||_chunkEAWYV5ZOcjs.a["cm-muted-primary"]}};
    color: ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    cursor: pointer;
    
    /* Transitions */
    transition: background-color 200ms ease-in-out;

    &:hover {
        background-color: ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
    }
    
    &:focus {
        outline: none;
        background-color: ${_chunkEAWYV5ZOcjs.a["cm-hover"]};
    }
`,x=_styled2.default.div`
    position: absolute;
    left: 20px;
    display: flex;
    align-items: center;
    justify-content: center;
`,g=_styled2.default.div`
    position: absolute;
    right: 20px;
    display: flex;
    align-items: center;
    justify-content: center;
`,h=_styled2.default.span`
    font-weight: 400;
    margin: 0 32px;
    color: ${e=>{var t,o;return((o=(t=e.appearance)==null?void 0:t.colors)==null?void 0:o.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
`;function R(){var a,c;let{step:e,appearance:t,setStep:o,setError:d}=_chunkHNUI5756cjs.a.call(void 0, );return e!=="initial"?null:_jsxruntime.jsxs.call(void 0, f,{appearance:t,onClick:()=>{o("web3"),d(null)},children:[_jsxruntime.jsx.call(void 0, x,{children:_jsxruntime.jsx.call(void 0, _chunkYCYSZLQ2cjs.a,{style:{height:"21px",width:"21px",color:(a=t==null?void 0:t.colors)==null?void 0:a.textPrimary}})}),_jsxruntime.jsx.call(void 0, h,{appearance:t,children:"Continue with a wallet"}),_jsxruntime.jsx.call(void 0, g,{children:_jsxruntime.jsx.call(void 0, _chunkCDQFTAM2cjs.a,{style:{height:"21px",width:"21px",color:(c=t==null?void 0:t.colors)==null?void 0:c.textSecondary}})}),_jsxruntime.jsx.call(void 0, _chunkS4BKL6MRcjs.a,{children:"Continue with a wallet"})]})}exports.a = R;
