// @crossmint/client-sdk-react-ui/dist/chunk-O3H2EYFA.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { newObj[key] = obj[key]; } } } newObj.default = obj; return newObj; } } function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunk3X24JZVGcjs = require('./chunk-3X24JZVG.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _react = require('react'); var n = _interopRequireWildcard(_react);var _reactdom = require('react-dom');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _reacttransitiongroup = require('react-transition-group');var _jsxruntime = require('react/jsx-runtime');var C=_styled2.default.div`
    position: fixed;
    inset: 0;
    z-index: 999;
    background-color: rgba(0, 0, 0, 0.5);
    backdrop-filter: blur(8px);
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 16px;
    
    /* Transition classes */
    &.dialog-enter {
        opacity: 0;
    }
    &.dialog-enter-active {
        opacity: 1;
        transition: opacity 100ms ease;
    }
    &.dialog-exit {
        opacity: 1;
    }
    &.dialog-exit-active {
        opacity: 0;
        transition: opacity 200ms ease;
    }
`,R=_styled2.default.div`
    ${_chunk3X24JZVGcjs.a}
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif;
    
    /* Modal styling */
    position: relative;
    background-color: ${t=>{var e,a;return((a=(e=t.appearance)==null?void 0:e.colors)==null?void 0:a.background)||_chunkEAWYV5ZOcjs.a["cm-background-primary"]}};
    border-radius: ${t=>{var e;return((e=t.appearance)==null?void 0:e.borderRadius)||"16px"}};
    box-shadow: 0 20px 25px -5px rgba(0, 0, 0, 0.1), 0 10px 10px -5px rgba(0, 0, 0, 0.04);
    width: 100%;
    max-height: 90vh;
    padding: 32px;
    overflow-y: auto;
    
    /* Focus management */
    &:focus {
        outline: none;
    }

    /* Desktop styles and animations */
    @media (min-width: 480px) {
        max-width: 448px;
        border-radius: ${t=>{var e;return((e=t.appearance)==null?void 0:e.borderRadius)||"24px"}};
        padding: 40px 40px 30px;
        
        /* Desktop transition classes */
        .dialog-enter & {
            transform: scale(0.95);
            opacity: 0;
        }
        .dialog-enter-active & {
            transform: scale(1);
            opacity: 1;
            transition: transform 200ms ease, opacity 100ms ease;
        }
        .dialog-exit & {
            transform: scale(1);
            opacity: 1;
        }
        .dialog-exit-active & {
            transform: scale(0.95);
            opacity: 0;
            transition: transform 200ms ease, opacity 200ms ease;
        }
    }

    /* Mobile styles and animations */
    @media (max-width: 479px) {
        position: fixed;
        left: 0;
        right: 0;
        bottom: 8px;
        width: calc(100% - 16px);
        max-height: calc(100vh - 18px);
        border-radius: ${t=>{var e;return((e=t.appearance)==null?void 0:e.borderRadius)||"36px 36px 50px 50px"}};
        padding: 40px 30px;
        margin: 0 auto;
        
        /* Mobile transition classes */
        .dialog-enter & {
            transform: translateY(100%);
        }
        .dialog-enter-active & {
            transform: translateY(0);
            transition: transform 200ms ease;
        }
        .dialog-exit & {
            transform: translateY(0);
        }
        .dialog-exit-active & {
            transform: translateY(100%);
            transition: transform 200ms ease;
        }
    }
`,k=_styled2.default.button`
    ${_chunk3X24JZVGcjs.a}
    
    /* Close button styling */
    position: absolute;
    top: 28px;
    right: 28px;
    width: 32px;
    height: 32px;
    border-radius: 6px;
    border: none;
    z-index: 1000;
    background-color: transparent;
    color: ${t=>{var e,a,i;return(i=(a=(e=t.appearance)==null?void 0:e.colors)==null?void 0:a.textPrimary)!=null?i:_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    cursor: pointer;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 0.15s ease;
    
    &:hover {
        background-color: ${t=>{var e,a;return((a=(e=t==null?void 0:t.appearance)==null?void 0:e.colors)==null?void 0:a.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
        color: ${t=>{var e,a,i;return(i=(a=(e=t.appearance)==null?void 0:e.colors)==null?void 0:a.textPrimary)!=null?i:_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    }
    
    &:focus {
        outline: none;
        background-color:  ${t=>{var e,a;return((a=(e=t==null?void 0:t.appearance)==null?void 0:e.colors)==null?void 0:a.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
        box-shadow: 0 0 0 2px ${t=>{var e,a;return((a=(e=t.appearance)==null?void 0:e.colors)==null?void 0:a.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}};
    }
    
    svg {
        width: 28px;
        height: 28px;
    }
`,w=_styled2.default.h2`
    ${_chunk3X24JZVGcjs.a}
    
    /* Title styling */
    display: block;
    font-size: 24px;
    font-weight: 600;
    line-height: 1.3;
    color: ${t=>{var e,a;return((a=(e=t.appearance)==null?void 0:e.colors)==null?void 0:a.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    margin: 0 0 2px 0;
`,S=_styled2.default.p`
    ${_chunk3X24JZVGcjs.a}
    
    /* Description styling */
    display: block;
    font-size: 16px;
    font-weight: 400;
    line-height: 1.5;
    color: ${t=>{var e,a;return((a=(e=t.appearance)==null?void 0:e.colors)==null?void 0:a.textSecondary)||_chunkEAWYV5ZOcjs.a["cm-text-secondary"]}};
    margin: 0 0 12px 0;
`,$=_styled2.default.div`
    /* Content wrapper */
    display: flex;
    flex-direction: column;
`,M= exports.a =({open:t,setDialogOpen:e,appearance:a,style:i,children:f,closeOnOverlayClick:x=!1,showCloseButton:y=!0})=>{let[u,g]=n.useState(!1),d=n.useRef(null);return n.useEffect(()=>{g(!0)},[]),n.useEffect(()=>(t?document.body.style.overflow="hidden":document.body.style.overflow="",()=>{document.body.style.overflow=""}),[t]),n.useEffect(()=>{let s=b=>{b.key==="Escape"&&t&&(e==null||e(!1))};return document.addEventListener("keydown",s),()=>document.removeEventListener("keydown",s)},[t,e]),u?_reactdom.createPortal.call(void 0, _jsxruntime.jsx.call(void 0, _reacttransitiongroup.CSSTransition,{nodeRef:d,in:t,timeout:200,classNames:"dialog",unmountOnExit:!0,mountOnEnter:!0,children:_jsxruntime.jsx.call(void 0, C,{ref:d,onClick:x?s=>{s.target===s.currentTarget&&(e==null||e(!1))}:void 0,children:_jsxruntime.jsxs.call(void 0, R,{appearance:a,role:"dialog","aria-modal":"true",style:_chunk2WSEHVNPcjs.a.call(void 0, {},i),children:[y&&_jsxruntime.jsx.call(void 0, k,{onClick:()=>e==null?void 0:e(!1),appearance:a,children:_jsxruntime.jsxs.call(void 0, "svg",{viewBox:"0 0 24 24",fill:"none",stroke:"currentColor",strokeWidth:"2",children:[_jsxruntime.jsx.call(void 0, "path",{d:"M18 6 6 18"}),_jsxruntime.jsx.call(void 0, "path",{d:"m6 6 12 12"})]})}),_jsxruntime.jsx.call(void 0, $,{children:f})]})})}),document.body):null},B= exports.b =({children:t,style:e,appearance:a})=>_jsxruntime.jsx.call(void 0, w,{style:e,appearance:a,children:t}),F= exports.c =({children:t,style:e,appearance:a})=>_jsxruntime.jsx.call(void 0, S,{style:e,appearance:a,children:t});exports.a = M; exports.b = B; exports.c = F;
