// @crossmint/client-sdk-react-ui/dist/chunk-DJZOP4MT.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunk3X24JZVGcjs = require('./chunk-3X24JZVG.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _react = require('react');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var L=_styled2.default.button`
    ${_chunk3X24JZVGcjs.a}
    
    /* Button styles */
    font-size: 14px;
    line-height: 1.3;
    text-align: center;
    margin-top: 8px;
    transition: opacity 200ms ease;
    border: none;
    background: none;
    border-radius: 4px;
    
    /* Conditional styling based on state */
    cursor: ${e=>e.canResend&&!e.isLoading?"pointer":"default"};
    opacity: ${e=>e.canResend&&!e.isLoading?1:.5};
    color: ${e=>{var t,n,s,i;return e.canResend&&!e.isLoading?((n=(t=e.appearance)==null?void 0:t.colors)==null?void 0:n.textLink)||"#1A73E8":((i=(s=e.appearance)==null?void 0:s.colors)==null?void 0:i.textSecondary)||"#67797F"}};
    
    /* Hover state for active button */
    &:hover {
        opacity: ${e=>e.canResend&&!e.isLoading?.8:.5};
    }

    /* Tab style for button (for accessibility) */
    &:focus-visible {
        outline: 2px solid ${e=>{var t,n;return((n=(t=e.appearance)==null?void 0:t.colors)==null?void 0:n.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}};
        outline-offset: 2px;
    }
    
    /* Disabled state */
    &:disabled {
        cursor: default;
    }
`;function S({initialSeconds:e,appearance:t,countdownText:n,countdownCompleteText:s,handleOnClick:i}){let[r,u]=_react.useState.call(void 0, e),[o,f]=_react.useState.call(void 0, !0),[y,C]=_react.useState.call(void 0, 0),[l,p]=_react.useState.call(void 0, !1);_react.useEffect.call(void 0, ()=>{let a;return r>0&&(f(!1),a=setInterval(()=>{u(c=>c<=1?(f(!0),0):c-1)},1e3)),()=>{a!=null&&clearInterval(a)}},[r]);let R=()=>{if(!o)return;let a=Math.min(e*Math.pow(2,y),240);u(a),C(c=>c+1)};return _jsxruntime.jsx.call(void 0, L,{onClick:()=>_chunk2WSEHVNPcjs.d.call(void 0, this,null,function*(){o&&r===0&&!l&&(p(!0),yield i(),R(),p(!1))}),disabled:!o||l,appearance:t,canResend:o,isLoading:l,children:o?s:n(r)})}exports.a = S;
