// @crossmint/client-sdk-react-ui/dist/chunk-6O6IJTQD.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkHNUI5756cjs = require('./chunk-HNUI5756.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var g=_styled2.default.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 8px 0px 12px;
`,e=_styled2.default.span`
    width: 100%;
    height: 1px;
    background-color: ${_chunkEAWYV5ZOcjs.a["cm-border"]};
`,x=_styled2.default.p`
    color: ${_chunkEAWYV5ZOcjs.a["cm-text-primary"]};
    font-size: 14px;
    font-weight: 400;
    margin: 1px;
    padding: 0 8px;
`;function v({appearance:o,text:d}){var n,l,s;let{step:f}=_chunkHNUI5756cjs.a.call(void 0, );return f!=="initial"?null:_jsxruntime.jsxs.call(void 0, g,{children:[_jsxruntime.jsx.call(void 0, e,{style:{backgroundColor:(n=o==null?void 0:o.colors)==null?void 0:n.border}}),d!=null?_jsxruntime.jsx.call(void 0, x,{style:{color:(l=o==null?void 0:o.colors)==null?void 0:l.textSecondary},children:d}):null,_jsxruntime.jsx.call(void 0, e,{style:{backgroundColor:(s=o==null?void 0:o.colors)==null?void 0:s.border}})]})}exports.a = v;
