// @crossmint/client-sdk-react-ui/dist/chunk-CD6E2BGD.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkS4BKL6MRcjs = require('./chunk-S4BKL6MR.cjs');var _chunkAHMPKIEYcjs = require('./chunk-AHMPKIEY.cjs');var _chunk3X24JZVGcjs = require('./chunk-3X24JZVG.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var x=_styled2.default.button`
    ${_chunk3X24JZVGcjs.a}
    
    /* Base button styles */
    position: relative;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    padding: 16px;
    font-size: 16px;
    min-height: 56px;
    border: none;
    border-radius: ${e=>{var o;return((o=e.appearance)==null?void 0:o.borderRadius)||"12px"}};
    background-color: ${e=>{var o,t;return((t=(o=e.appearance)==null?void 0:o.colors)==null?void 0:t.buttonBackground)||_chunkEAWYV5ZOcjs.a["cm-muted-primary"]}};
    color: ${e=>{var o,t;return((t=(o=e.appearance)==null?void 0:o.colors)==null?void 0:t.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    cursor: ${e=>e.isLoading?"not-allowed":"pointer"};
    
    /* Transitions */
    transition: background-color 200ms ease-in-out;
    
    /* Hover and focus states */
    &:hover {
        background-color: ${e=>{var o,t,n,i;return e.isLoading?((t=(o=e.appearance)==null?void 0:o.colors)==null?void 0:t.buttonBackground)||_chunkEAWYV5ZOcjs.a["cm-muted-primary"]:((i=(n=e.appearance)==null?void 0:n.colors)==null?void 0:i.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
    }
    
    &:focus {
        outline: none;
        background-color: ${e=>{var o,t;return((t=(o=e.appearance)==null?void 0:o.colors)==null?void 0:t.backgroundSecondary)||_chunkEAWYV5ZOcjs.a["cm-hover"]}};
    }
`,B=_styled2.default.div`
    position: absolute;
    left: 18px;
    display: flex;
    align-items: center;
    justify-content: center;
`,h=_styled2.default.span`
    font-weight: 400;
    color: ${e=>{var o,t;return((t=(o=e.appearance)==null?void 0:o.colors)==null?void 0:t.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
`;function I(C){var s=C,{icon:e,children:o,isLoading:t=!1,appearance:n,onButtonClick:i}=s,b=_chunk2WSEHVNPcjs.c.call(void 0, s,["icon","children","isLoading","appearance","onButtonClick"]);var u,d;return _jsxruntime.jsxs.call(void 0, x,_chunk2WSEHVNPcjs.b.call(void 0, _chunk2WSEHVNPcjs.a.call(void 0, {appearance:n,isLoading:t,onClick:t?void 0:i},b),{children:[_jsxruntime.jsx.call(void 0, B,{children:e}),t?_jsxruntime.jsx.call(void 0, _chunkAHMPKIEYcjs.a,{style:{color:(u=n==null?void 0:n.colors)==null?void 0:u.textSecondary,fill:(d=n==null?void 0:n.colors)==null?void 0:d.textPrimary}}):_jsxruntime.jsx.call(void 0, h,{appearance:n,children:o}),_jsxruntime.jsx.call(void 0, _chunkS4BKL6MRcjs.a,{children:o})]}))}exports.a = I;
