// @crossmint/client-sdk-react-ui/dist/chunk-OVJIGWD7.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true}); function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }var _chunkAHMPKIEYcjs = require('./chunk-AHMPKIEY.cjs');var _chunkEAWYV5ZOcjs = require('./chunk-EAWYV5ZO.cjs');var _chunk2WSEHVNPcjs = require('./chunk-2WSEHVNP.cjs');var _react = require('react');var _styled = require('@emotion/styled'); var _styled2 = _interopRequireDefault(_styled);var _jsxruntime = require('react/jsx-runtime');var b=_styled2.default.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    max-width: 448px;
    margin: 0 auto;
    border-radius: 12px;
`,h=_styled2.default.h2`
    margin: 6px 0 8px 0;
    font-size: 18px;
    font-weight: 600;
    color: ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
`,v=_styled2.default.div`
    margin-bottom: 32px;
    padding: 16px;
    border-radius: 12px;
    border: 1px solid ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.border)||_chunkEAWYV5ZOcjs.a["cm-border"]}};
`,I=_styled2.default.div`
    display: flex;
    align-items: center;
`,B=_styled2.default.div`
    flex-shrink: 0;
    margin-right: 12px;
`,w=_styled2.default.span`
    color: ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    font-size: 16px;
    overflow-wrap: anywhere;
`,k=_styled2.default.div`
    color: ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.danger)||_chunkEAWYV5ZOcjs.a["cm-danger"]}};
    margin-bottom: 16px;
    font-size: 14px;
`,S=_styled2.default.div`
    display: flex;
    gap: 12px;
`,x=_styled2.default.button`
    flex: 1;
    padding: 16px 24px;
    border-radius: 9999px;
    text-align: center;
    font-weight: 500;
    font-size: 16px;
    cursor: pointer;
    transition: opacity 200ms ease;
    border: none;
    &:disabled {
        cursor: not-allowed;
    }
`,U=_styled2.default.call(void 0, x)`
    background: transparent;
    border: 1px solid ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.border)||_chunkEAWYV5ZOcjs.a["cm-border"]}};
    color: ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.textPrimary)||_chunkEAWYV5ZOcjs.a["cm-text-primary"]}};
    &:hover:not(:disabled) {
        opacity: 0.8;
    }
`,$=_styled2.default.call(void 0, x)`
    background-color: ${e=>{var o,n;return((n=(o=e.appearance)==null?void 0:o.colors)==null?void 0:n.accent)||_chunkEAWYV5ZOcjs.a["cm-accent"]}};
    color: white;
    opacity: ${e=>e.isLoading?.7:1};
    &:hover:not(:disabled) {
        opacity: 0.9;
    }
`,L=_styled2.default.div`
    display: flex;
    justify-content: center;
    align-items: center;
`;function R({contactInfo:e,icon:o,onConfirm:n,onCancel:u,appearance:t}){var m;let[c,d]=_react.useState.call(void 0, !1),[p,l]=_react.useState.call(void 0, "");return _jsxruntime.jsxs.call(void 0, b,{children:[_jsxruntime.jsx.call(void 0, h,{appearance:t,children:"Send authorization code to"}),_jsxruntime.jsx.call(void 0, v,{appearance:t,children:_jsxruntime.jsxs.call(void 0, I,{children:[_jsxruntime.jsx.call(void 0, B,{children:o}),_jsxruntime.jsx.call(void 0, w,{appearance:t,children:e})]})}),p&&_jsxruntime.jsx.call(void 0, k,{appearance:t,children:p}),_jsxruntime.jsxs.call(void 0, S,{children:[_jsxruntime.jsx.call(void 0, U,{onClick:u,appearance:t,children:"Cancel"}),_jsxruntime.jsx.call(void 0, $,{onClick:()=>_chunk2WSEHVNPcjs.d.call(void 0, this,null,function*(){d(!0),l("");try{yield n(e)}catch(y){console.error("Error sending authorization code:",y),l("Failed to send code. Please try again.")}finally{d(!1)}}),disabled:c,isLoading:c,appearance:t,children:c?_jsxruntime.jsx.call(void 0, L,{children:_jsxruntime.jsx.call(void 0, _chunkAHMPKIEYcjs.a,{style:{color:((m=t==null?void 0:t.colors)==null?void 0:m.background)||_chunkEAWYV5ZOcjs.a["cm-background-primary"]}})}):"Send code"})]})]})}exports.a = R;
